docs/source/tutorial/understanding_evaluation.rst

```diff
@@ -0,0 +1,69 @@
+Understanding the Evaluation
+============================
+This part of the tutorial is aimed to help you understand the evaluation of knowledge graph embeddings.
+In particular it explains rank-based evaluation metrics reported in :class:`pykeen.evaluation.RankBasedMetricResults`.
+
+Knowledge graph embedding are usually evaluated on the task of link prediction. To this end, an evaluation set of
+triples :math:`\mathcal{T}_{eval} \subset \mathcal{E} \times \mathcal{R} \times \mathcal{E}` is provided, and for each
+triple :math:`(h, r, t) \in \mathcal{T}_{eval}` in this set, we solve two tasks:
+
+* In the *right-side* prediction task, we are given a par of head entity and relation and aim to predict the tail,
+  i.e. :math:`(h, r, ?)`. To this end, we use the knowledge graph embedding model to *score* each of the possible
+  choices :math:`(h, r, e)` for :math:`e \in \mathcal{E}`. Higher scores indicate higher plausibility.
+* Analogously, in the *left-side* prediction task, we are provided a pair of relation and tail entity and aim to
+  predict the head, i.e. :math:`(?, r, t)`. Again, each possible choice :math:`(e, r, t)` for
+  :math:`e \in \mathcal{E}` is scored according to the knowledge graph embedding model.
+
+.. note ::
+    Practically, many embedding models allow fast computation of all scores :math:`(e, r, t)` for all
+    :math:`e \in \mathcal{E}`, than just passing the triples through the model's score function. As an example,
+    consider DistMult with the score function
+    :math:`score(h,r,t)=\sum_{i=1}^d \mathbf{h}_i \cdot \mathbf{r}_i \cdot \mathbf{t}_i`. Here, we can score all
+    entities as candidate heads for a given tail and relation by first computing the element-wise product of tail and
+    relation, and then performing a matrix multiplication with the matrix of all entity embeddings.
+    # TODO: Link to section explaining this concept.
```

Comment:
@lvermue (@cthoyt ) I think we should describe this somewhere in a dedicated section of the documentation where we talk about performance optimizations.

Reply:
agreed! I made a placeholder for this in https://github.com/pykeen/pykeen/issues/55